lua/projectlaunch.lua

```diff
@@ -19,6 +21,19 @@ M.show_next = term.show_next
 
 M.setup = function(opts)
 	options.override(opts)
+
+	if options.get().auto_reload_config == true then
+		-- reload config when reload a saved session
+		api.nvim_create_autocmd("SessionLoadPost", {
+			callback = config.reload_config,
+		})
+
+		-- reload config when updated custom config
+		api.nvim_create_autocmd(
+			"BufWritePost",
+			{ pattern = options.get().config_path, callback = config.reload_config }
+		)
+	end
```

Comment:
What do you think of moving the check into `reload_config`, and moving the autocmds out of `setup`? Even though `auto_reload_config` is now `true` by default, having it in `setup` means the `autocmd`s for it aren't created unless the user calls `setup()` even if they don't override any settings, so if the user doesn't care to override any of the default settings they wouldn't get this automatically.

Something like this maybe?

```lua
local util = require("projectlaunch.util")

local function reload_config()
	if options.get().auto_reload_config then
		config.reload_config()
	end
end

-- reload config when reload a saved session
api.nvim_create_autocmd("SessionLoadPost", {
	group = util.augroup,
	callback = reload_config,
})

-- reload config when updated custom config
api.nvim_create_autocmd("BufWritePost", {
	group = util.augroup,
	pattern = options.get().config_path,
	callback = reload_config,
})

M.setup = function(opts)
	options.override(opts)
end

```

Reply:
Thanks for your suggestion, and I moved autocmds into `config.lua`.

New commit was pushed to my `main` branch.